app/src/test/java/org/simple/clinic/setup/SetupActivityInitTest.kt

```diff
@@ -0,0 +1,23 @@
+package org.simple.clinic.setup
+
+import com.spotify.mobius.test.FirstMatchers.hasEffects
+import com.spotify.mobius.test.FirstMatchers.hasModel
+import com.spotify.mobius.test.InitSpec
+import com.spotify.mobius.test.InitSpec.assertThatFirst
+import org.junit.Test
+
+class SetupActivityInitTest {
+
+  @Test
+  fun `when the screen is created, the user details must be loaded`() {
```

Comment:
What if the details are already loaded, in case of a restoration?

Reply:
It doesn't really matter in this case, because it is okay to fetch user details again.

Reply:
Currently, this only fetches whether the user has done the onboarding or not. Later on, this will get more information like whether a user exists, or whether they have fetched a country. I will update this then.